docs/operators/jsonParser.md

```diff
@@ -0,0 +1,31 @@
+Json Parser
+=============
+
+## Operator Objective
+Purpose of this operator is to parse JSON records and construct a Plain Old Java Object ("POJO") out of it. The operator also emits each record as JSONObject. User can also provide a schema describing JSON data to validate incoming JSON records. Valid records will be emitted as POJO / JSONOnject while invalid ones are emitted on error port with error message.
+
+Json Parser does not hold any state and is **idempotent**, **fault-tolerance** & **statically/dynamically partitionable**.
+
+## Properties, Attributes and Ports
+### <a name="props"></a>Properties of Json Parser
+| **Property** | **Description** | **Type** | **Mandatory** | **Default Value** |
+| -------- | ----------- | ---- | ------------------ | ------------- |
+| *jsonSchema* | [Schema](http://json-schema.org/)  describing JSON data | String | NO | N/A |
+
+### Platform Attributes that influences operator behavior
+| **Attribute** | **Description** | **Type** | **Mandatory** |
+| -------- | ----------- | ---- | ------------------ |
+| *out.TUPLE_CLASS* | TUPLE_CLASS attribute on output port which tells operator the class of POJO which need to be emitted | Class| Yes |
+
+
+### Ports
+| **Port** | **Description** | **Type** | **Mandatory** |
+| -------- | ----------- | ---- | ------------------ |
+| *out* | Valid Tuples that are emitted as pojo | Object (POJO) | Yes |
+| *parsedOutput* | Valid Tuples that are emitted as JSONObject | JSONObject | No |
+| *err* | Invalid Tuples are emitted with error message | KeyValPair <String, String\> | No |
+
```

Comment:
The `out` port is described as mandatory but is not annotated that way in the base class -- in fact the base class treats it as optional. This sort of discrepancy is likely to confuse the reader unless we provide a clear explanation. Also add an explicit statement that if the error port is not connected, erroneous tuples are discarded and, similarly, if the parsedOutput port is not connected, JSONObjects are not created.